unit_test/tstForce.hpp

```diff
@@ -1193,10 +1202,239 @@ TEST( TEST_CATEGORY, test_force_lps_multi )
+                                 CabanaPD::NoFracture{}, delta, K, G, 1 );
+    CabanaPD::ForceModel model2( model_type{}, delta, K, G, G0, 1 );
+    CabanaPD::ForceModel model3( model_type{}, delta, K, G, G0, 1 );
+    CabanaPD::ForceModel model4( model_type{}, delta, K, G, G0, 1 );
+
+    auto particles = createParticles( model_type{}, LinearTag{}, dx, 0.1 );
+    auto models = CabanaPD::createMultiForceModel(
+        particles, CabanaPD::AverageTag{}, model1, model2, model3, model4 );
+
+    static_assert(
+        std::is_same_v<model_type, typename decltype( models )::model_type> );
+    static_assert( std::is_same_v<typename model_type::base_model,
+                                  typename decltype( models )::base_model> );
+    static_assert( std::is_same_v<CabanaPD::Fracture,
+                                  typename decltype( models )::fracture_type> );
+    static_assert( std::is_same_v<CabanaPD::TemperatureIndependent,
+                                  typename decltype( models )::thermal_type> );
+
     Inputs<model_type> inputs{ delta, K, G, 0.1, 2.1 };
     testForce( models, dx, m, LinearTag{}, inputs );
     inputs.update( 0.01 );
     testForce( models, dx, m, QuadraticTag{}, inputs );
 }
 
+namespace
+{
+struct FirstTestTag
+{
+};
+struct BaseModelDummyOne
+{
+};
+struct BaseModelDummyTwo
+{
+};
+
+template <typename BaseModel>
+struct TestModelType
+{
+    using base_model = BaseModel;
+    KOKKOS_INLINE_FUNCTION auto operator()( int index ) const { return index; }
+};
+
+template <typename ModelType, typename BaseModel, typename FractureType,
+          typename ThermalType>
+struct TestModel
+{
+    using model_type = ModelType;
+    using base_model = BaseModel;
+    using fracture_type = FractureType;
+    using thermal_type = ThermalType;
+
+    int expectedTypeI;
+    int expectedTypeJ;
+    int expectedParameter;
+
+    double delta = 0.1;
+
+    KOKKOS_INLINE_FUNCTION bool operator()( FirstTestTag, const int typeI,
+                                            const int typeJ,
+                                            const int inputValue ) const
+    {
+        if ( inputValue != expectedParameter )
+            Kokkos::abort( "Wrong inputValue in operator with FirstTestTag" );
+        if ( typeI != expectedTypeI )
+            Kokkos::abort( "Wrong typeI in operator with FirstTestTag" );
+        if ( typeJ != expectedTypeJ )
+            Kokkos::abort( "Wrong typeJ in operator with FirstTestTag" );
+
+        return true;
+    }
+
+    KOKKOS_INLINE_FUNCTION bool operator()( FirstTestTag,
+                                            CabanaPD::MultiMaterial,
+                                            const int typeI, const int typeJ,
+                                            const int inputValue ) const
+    {
+        return this->operator()( FirstTestTag(), typeI, typeJ, inputValue );
+    }
+};
+} // namespace
+
+TEST( TEST_CATEGORY, test_forceModelsMulti_binary )
+{
```

Comment:
I tried to develop a test that only tests the implementation in `ForceModelsMulti` with "forces" that check that we get the correct model (the recursion stops at the correct index in parameter pack) for the respective types of particles